code/modules/holomap/portable_map_reader.dm

```diff
@@ -0,0 +1,64 @@
+
+/obj/item/portable_map_reader
+	name = "portable map reader"
+	desc = "Displays a map of the local space, as well as any marked points of interest."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "depthscanner"
+
+	/// The zlevel that this reader is spawned on.
+	/// It only shows that zlevel (and any zlevels that zlevel is connected to).
+	/// It will not show other zlevels, if the user were to transport it elsewhere on a shuttle.
+	var/starting_z_level = null
```

Comment:
Why not just take the connected zlevels in lateinitialize and store that, instead of recomputing it everytime on ui_data?